hs_composite_resource/models.py

```diff
@@ -138,35 +362,73 @@ def supports_logical_file(self):
         return True
 
     def get_metadata_xml(self, pretty_print=True, include_format_elements=True):
-        from lxml import etree
-
         # get resource level core metadata as xml string
         # for composite resource we don't want the format elements at the resource level
         # as they are included at the aggregation map xml document
-        xml_string = super(CompositeResource, self).get_metadata_xml(pretty_print=False,
+        xml_string = super(CompositeResource, self).get_metadata_xml(pretty_print=True,
                                                                      include_format_elements=False)
+        return xml_string
 
-        # create an etree xml object
-        RDF_ROOT = etree.fromstring(xml_string)
-
-        return etree.tostring(RDF_ROOT, encoding='UTF-8', pretty_print=pretty_print).decode()
-
-    def _recreate_fileset_xml_docs(self, folder):
-        """Recreates xml files for all fileset aggregations that exist under the path 'folder'
-        as well as for any parent fileset that may exist relative to path 'folder'
+    def _recreate_nested_aggr_xml_docs(self, folder, nested_aggr):
+        """Recreates xml files for all fileset or model instance aggregations that exist under the path 'folder'
+        as well as for any parent fileset/model instance that may exist relative to path 'folder'
         """
+        if nested_aggr == 'fileset':
+            nested_aggr_set = self.filesetlogicalfile_set
+            aggr_in_path_func = self.get_fileset_aggregation_in_path
+        else:
+            # create xml files for all model instance aggregation that may exist under *folder*
+            nested_aggr_set = self.modelinstancelogicalfile_set
+            aggr_in_path_func = self.get_model_aggregation_in_path
 
-        filesets = self.filesetlogicalfile_set.filter(folder__startswith=folder)
-        for fs in filesets:
-            fs.create_aggregation_xml_documents()
+        nested_aggrs = nested_aggr_set.filter(folder__startswith=folder)
```

Comment:
Why not just filter aggregations by resource and folder?  I think it makes sense to avoid treating different logical file types differently when in the resource.